File: app/src/main/java/com/boardgamegeek/pref/BuddyPreference.java
package com.boardgamegeek.pref;

import android.content.Context;
import android.content.Intent;
import android.preference.Preference;
import android.util.AttributeSet;

import com.boardgamegeek.ui.BuddyActivity;

public class BuddyPreference extends Preference {
	public BuddyPreference(Context context, AttributeSet attrs) {
		super(context, attrs);
		Intent intent = BuddyActivity.createIntent(context, getSummary().toString(), null);
		setIntent(intent);
	}
}

File: app/src/main/java/com/boardgamegeek/pref/BuddyPreference.kt
package com.boardgamegeek.pref

import android.content.Context
import android.util.AttributeSet
import androidx.preference.Preference
import com.boardgamegeek.ui.BuddyActivity

class BuddyPreference(context: Context, attrs: AttributeSet) : Preference(context, attrs) {
    init {
        intent = BuddyActivity.createIntent(context, summary.toString(), null)
    }
}
